src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -73,8 +73,8 @@ public static boolean unlock(final Device dev) {
         FourBitTwoDisclosureDeviceUnlocker.dev = dev;
         boolean isUnlocked = doSpin();
         List<CharSequence> perms = getPermutations();
-        int n = NUM_OF_BITS;
-        while ((!isUnlocked) && (n > 0)) {
+        int n = 1;
```

Comment:
This is where the increase in efficiency is.

Instead of trying the maximum number of spins and approaching 1, we start at 1 and approach the maximum number of spins.